### PR TITLE
Enhancing activity list

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
While debugging #6477  I enhanced it…

- show correct placeholder
- click on preview open file

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [ ] Tests written, or not not needed

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/adapter/ActivityListAdapter.java [hunk 1]
```diff
@@ -256,23 +259,34 @@ public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int positi
         }
     }
 
-    private ImageView createThumbnailNew(PreviewObject previewObject) {
+    private ImageView createThumbnailNew(PreviewObject previewObject, ArrayList<RichObject> richObjectList) {
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/adapter/ActivityListAdapter.java:262]: @tobiasKaminsky should use `List` instead

File: src/main/java/com/owncloud/android/ui/adapter/ActivityListAdapter.java [hunk 2]
```diff
@@ -256,23 +259,34 @@ public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int positi
...
         LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(px, px);
         params.setMargins(10, 10, 10, 10);
         ImageView imageView = new ImageView(context);
         imageView.setLayoutParams(params);
 
+        for (RichObject object : richObjectList) {
+            if (Integer.parseInt(object.getId()) == previewObject.getFileId()) {
+                imageView.setOnClickListener(v -> activityListInterface.onActivityClicked(object));
+                break;
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/adapter/ActivityListAdapter.java:271]: why the `break`?

**Comment by AndyScherzinger** [comment]
should be backported imho 👍

File: src/main/java/com/owncloud/android/ui/adapter/ActivityListAdapter.java (hunk 2 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/adapter/ActivityListAdapter.java:271]: https://github.com/nextcloud/activity/blob/master/docs/endpoint-v2.md#example

We are using the PreviewObject, which does not contain the remote path.
So we cannot handle click operation with only this info. (to open a file we currently need the remote path. We could of course also write a function that retrieves first the remote path for a given file from server and use this then).

As this info is already in subject_rich  list, we search for matching info. As there is no direct relation between those (not via index), we need to use fileID and compare it with id.
As there can only be one match, we can do a break, if we found this.

createThumbnailNew(…) creates then one ImageView with corresponding click listener.
For next preview object creatThumbnailNew(…) will be called again and thus do the same.

**Pull request merged**